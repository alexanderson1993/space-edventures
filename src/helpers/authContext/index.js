import React from "react";

const AuthContext = React.createContext({
  user: null,
  login: () => {},
  logout: () => {}
});

export default AuthContext;
