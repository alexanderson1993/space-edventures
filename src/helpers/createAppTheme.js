import { lighten, darken } from "polished";

const generateColor = color => ({
  base: color,
  light: lighten(0.2, color),
  dark: darken(0.2, color)
});

const generateBackground = color => ({
  level0: color,
  level1: lighten(0.015, color),
  level2: lighten(0.03, color),
  level3: lighten(0.045, color)
});

export default (theme = {}) => ({
  ...theme,
  color: {
    primary: generateColor(theme.colorPrimary || "#30fffe"),
    header: generateColor(theme.colorHeader || "#a1ecfb"),
    control: generateColor(theme.colorControl || "#acf9fb"),
    ...theme.color
  },
  background: {
    primary: generateBackground(theme.backgroundPrimary || "#031212"),
    ...theme.background
  }
});
