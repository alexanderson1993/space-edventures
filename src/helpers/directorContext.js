import React from "react";
export const DirectorContext = React.createContext({});
