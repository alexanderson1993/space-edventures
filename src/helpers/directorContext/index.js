import React, { useContext, useMemo } from "react";
import propTypes from "prop-types";
import AuthContext from "../authContext";
import { Query } from "react-apollo";
import graphQLHelper from "../graphQLHelper";
import CENTER_DIRECTOR from "./centerDirector.graphql";
import { Match } from "@reach/router";
export const DirectorContext = React.createContext({});

const Provider = ({ me = {}, children }) => {
  const value = useMemo(() => me, [me]);
  return (
    <DirectorContext.Provider value={value}>
      {children}
    </DirectorContext.Provider>
  );
};
export const DirectorProvider = ({ children }) => {
  const { user } = useContext(AuthContext);
  return (
    <Match path="/director/:centerId/*">
      {({ match }) => (
        <Query
          query={CENTER_DIRECTOR}
          variables={{
            centerId: match && match.centerId,
            id: user && user.id
          }}
          skip={!user || !user.id}
        >
          {graphQLHelper(({ me }) => (
            <Provider me={me} children={children} />
          ))}
        </Query>
      )}
    </Match>
  );
};

DirectorProvider.propTypes = {
  children: propTypes.node
};
