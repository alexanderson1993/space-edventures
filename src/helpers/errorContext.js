import React, { useReducer, useMemo } from "react";
export const ErrorContext = React.createContext();

const initialState = { error: false };
function reducer(state, action) {
  switch (action.type) {
    case "errored":
      return {
        error: true,
        errorData: action.errorData,
        errorInfo: action.errorInfo
      };
    case "cleared":
      return { error: false, errorData: null, errorInfo: null };
    default:
      throw new Error();
  }
}

export const ErrorProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const value = useMemo(() => ({ state, dispatch }), [state]);
  return (
    <ErrorContext.Provider value={value}>{children}</ErrorContext.Provider>
  );
};
