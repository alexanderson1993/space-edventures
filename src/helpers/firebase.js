import { firebase } from "@firebase/app";
import "@firebase/auth";

var config = {
  apiKey: process.env.GATSBY_FIREBASE_API_KEY,
  databaseURL: process.env.GATSBY_FIREBASE_DATABASE_URL,
  storageBucket: process.env.GATSBY_FIREBASE_STORAGE_BUCKET,
  authDomain: process.env.GATSBY_FIREBASE_AUTH_DOMAIN,
  messagingSenderId: process.env.GATSBY_FIREBASE_MESSAGE_SENDER_ID,
  projectId: process.env.GATSBY_FIREBASE_PROJECT_ID
};
let authItem;
let baseAuthItem;
if (typeof window !== "undefined") {
  firebase.initializeApp(config);
  authItem = firebase.auth();
  baseAuthItem = firebase.auth;
}
export const auth = authItem;
export const baseAuth = baseAuthItem;
