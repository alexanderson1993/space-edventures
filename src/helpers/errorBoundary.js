import React, { Component } from "react";
import { Button } from "../components";
import styled from "@emotion/styled";
import { ErrorContext } from "./errorContext";

const Container = styled("div")`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
export default class ErrorBoundary extends Component {
  static contextType = ErrorContext;
  componentDidCatch(error, errorInfo) {
    // Catch errors in any components below and re-render with error message
    this.context.dispatch({ type: "errored", errorData: error, errorInfo });
  }
  refresh() {
    if (typeof window !== "undefined") {
      localStorage.clear();
      window.location.reload();
    }
  }
  render() {
    if (this.context.state.error) {
      // Error path
      if (this.props.render) return this.props.render;
      return (
        <div className="error-boundary">
          <Container>
            <h2>Error in the Space EdVentures app.</h2>
            <Button onClick={this.refresh}>Refresh</Button>

            <h3>
              If you know what you are looking for, you can check out what went
              wrong.
            </h3>
            <details style={{ whiteSpace: "pre-wrap" }}>
              {this.context.state.errorData &&
                this.context.state.errorData.toString()}
              <br />
              {this.context.state.errorInfo.componentStack}
            </details>
          </Container>
        </div>
      );
    }
    // Normally, just render children
    return this.props.children;
  }
}
