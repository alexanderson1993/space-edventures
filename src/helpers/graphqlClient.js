import { ApolloClient } from "apollo-client";
import { InMemoryCache } from "apollo-cache-inmemory";
import { onError } from "apollo-link-error";
import { setContext } from "apollo-link-context";
import { ApolloLink } from "apollo-link";
import { createUploadLink } from "apollo-upload-client";
import { ApolloProvider } from "react-apollo";
import { ApolloProvider as ApolloHooksProvider } from "react-apollo-hooks";

import fetch from "isomorphic-fetch";

import { auth } from "./firebase";
import React, { useContext } from "react";
import AuthContext from "./authContext";

const uri =
  process.env.NODE_ENV === "production"
    ? process.env.GATSBY_GRAPHQL_URL
    : "http://localhost:5000/space-edventures-beta/us-central1/api/graphql";

const AuthLink = setContext(async (_, { headers }) => {
  // get the authentication token from local storage if it exists
  try {
    const token = await auth.currentUser.getIdToken();
    // return the headers to the context so httpLink can read them
    return {
      headers: {
        ...headers,
        authorization: token ? `Bearer ${token}` : ""
      }
    };
  } catch (err) {
    return headers;
  }
});

// TODO: Add sentry event tracking to this when Sentry eventually gets added.
const ErrorLink = onError(({ graphQLErrors, networkError }) => {
  // if (graphQLErrors)
  //   graphQLErrors.map(({ message, locations, path }) =>
  //     // console.error(
  //     //   `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
  //     // )
  //   );
  // if (networkError) console.error(`[Network error]: ${networkError}`);
});

const DataLink = createUploadLink({
  uri,
  credentials: "same-origin"
});

export const client = new ApolloClient({
  link: ApolloLink.from([AuthLink, ErrorLink, DataLink]),
  cache: new InMemoryCache(),
  fetch: typeof window !== "undefined" ? window.fetch : fetch
});

export default ({ children }) => {
  // const { loading } = useContext(AuthContext);
  // if (!loading)
  return (
    <ApolloProvider client={client}>
      <ApolloHooksProvider client={client}>{children}</ApolloHooksProvider>
    </ApolloProvider>
  );
  // return null;
};
