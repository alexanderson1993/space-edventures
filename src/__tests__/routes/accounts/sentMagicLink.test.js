import React from "react";
import { render } from "react-testing-library";
import SentMagicLink from "../../../pages/accounts/sentMagicLink";
import { ErrorProvider } from "../../../helpers/errorContext";

test("sentMagicLink", () => {
  const { container } = render(
    <ErrorProvider>
      <SentMagicLink />
    </ErrorProvider>
  );
  expect(container).toMatchSnapshot();
});
