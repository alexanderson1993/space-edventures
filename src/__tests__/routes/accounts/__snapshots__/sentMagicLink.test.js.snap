// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sentMagicLink 1`] = `
.emotion-0 {
  margin: 0 auto;
  width: 960px;
  max-width: 80%;
  padding: 0px 1.0875rem 1.45rem;
  padding-top: 1.45rem;
  margin-bottom: 10%;
}

@media only screen and (max-width:600px) {
  .emotion-0 {
    padding: 0px 0.5875rem 1.45rem;
    max-width: 98%;
  }
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <h1>
      Magic Link Sent
    </h1>
    <p
      data-testid="magic-link-text"
    >
      Check your inbox! We sent an email to 
      . Click the link in that email to sign in to Space EdVentures.
    </p>
  </div>
</div>
`;
